scripts/endpoints-checker.sh

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env bash
+
+set -uo pipefail
+${TRACE:+set -x}
+
+CHAIN_FILE=$1
+THRESHOLD=${THRESHOLD:-30}
+TIMEOUT=${TIMEOUT:-5}
+
+is_synced() {
+    local block_time=$1
+    if [[ -z $block_time ]]; then
+        echo "Error: Block time for endpoint is empty" >&2
+        exit 1
+    fi
+
+    diff=$(jq '(now - (.time | split(".")[0] + "Z" | fromdate)) | trunc' <<< $block_time)
+    echo "Latest block is from $diff seconds ago" >&2
+    if [[ $diff -gt $THRESHOLD ]]; then
+        echo "Error: Endpoint is not synced" >&2
+        exit 1
+    fi
+}
+
+check_rpc() {
+    local endpoint=$1
+    echo "Checking RPC endpoint $endpoint" >&2
+    block_time=$(curl -f -s -S --max-time $TIMEOUT $endpoint/status | jq '{time: .result.sync_info.latest_block_time}')
+    is_synced "$block_time"
+}
+
+check_rest() {
+    local endpoint=$1
+    echo "Checking REST endpoint $endpoint" >&2
+    block_time=$(curl -f -s -S --max-time $TIMEOUT $endpoint/blocks/latest | jq '{time: .block.header.time}')
```

Comment:
Id suggest adding also `--connect-timeout $TIMEOUT` to set connection timeout.
Max time is for transfer timeouts. Both are needed in my experience